src/server/models/WeatherData.js

```diff
@@ -0,0 +1,86 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const database = require('./database');
+const { log } = require('../log');
+const moment = require('moment');
+
+const sqlFile = database.sqlFile;
+
+class WeatherData {
+	/**
+	 * Constructor without the id sine it's not in the schema
+	 * @param weatherLocationId The weather location id
+	 * @param {Moment} startTime The weather data's startTime
+	 * @param {Moment} endTime The weather data's endTime
+	 * @param temperature The actual weather data (temperature)
+	 */
+	constructor(weatherLocationId, startTime, endTime, temperature) {
+		this.weatherLocationId = weatherLocationId;
+		this.startTime = startTime;
+		this.endTime = endTime;
+		this.temperature = temperature;
+	}
+
+	/**
+	 * Creates a new WeatherData from the data in a row.
+	 * @param {*} row The row from which the WeatherData is to be created.
+	 * @returns The new WeatherData object.
+	 */
+	static mapRow(row) {
+		return new WeatherData(row.weatherLocationId, row.startTime, row.endTime, row.temperature);
+	}
+
+	/**
+	  * Returns a promise to create the weather data table
+	  * @param conn is the connection to use.
+	  * @returns {Promise.<>}
+	  */
+	static createTable(conn) {
+		return conn.none(sqlFile('weather_data/create_weather_data_table.sql'));
+	}
+
+	/**
+	  * Returns a promise to retrieve the weather data by the primary key
+	  * @param {*} conn The connection to use.
+	  * @returns {Promise.<WeatherData>}
+	  */
+	// TODO: needs model and db get_weather_data
+	static async getWeatherData(conn) {
+		const rows = await conn.one(sqlFile('weather_data/get_weather_data_by_id_and_start_time.sql'));
```

Comment:
This isn't passing the needed values. See the sql for thoughts on how this should work.